drivers/gpio/gpio_intel_ehl.c

```diff
@@ -0,0 +1,595 @@
+
+	port_get_raw(dev, mask, &port_val, true);
+
+	port_val = (port_val & ~mask) | (mask & value);
+
+	port_set_raw(dev, mask, port_val);
+
+	return 0;
+}
+
+static int gpio_intel_ehl_port_set_bits_raw(const struct device *dev,
+					    uint32_t mask)
+{
+	return gpio_intel_ehl_port_set_masked_raw(dev, mask, mask);
+}
+
+static int gpio_intel_ehl_port_clear_bits_raw(const struct device *dev,
+					      uint32_t mask)
+{
+	return gpio_intel_ehl_port_set_masked_raw(dev, mask, 0);
+}
+
+static int gpio_intel_ehl_port_toggle_bits(const struct device *dev,
+					   uint32_t mask)
+{
+	uint32_t port_val;
+
+	port_get_raw(dev, mask, &port_val, true);
+
+	port_val ^= mask;
+
+	port_set_raw(dev, mask, port_val);
+
+	return 0;
+}
+
+static int gpio_intel_ehl_port_get_raw(const struct device *dev,
+				       uint32_t *value)
+{
+	return port_get_raw(dev, 0xFFFFFFFF, value, false);
+}
+
+static const struct gpio_driver_api gpio_intel_ehl_api = {
+	.pin_configure = gpio_intel_ehl_config,
+	.manage_callback = gpio_intel_ehl_manage_callback,
+	.port_get_raw = gpio_intel_ehl_port_get_raw,
+	.port_set_masked_raw = gpio_intel_ehl_port_set_masked_raw,
+	.port_set_bits_raw = gpio_intel_ehl_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_intel_ehl_port_clear_bits_raw,
+	.port_toggle_bits = gpio_intel_ehl_port_toggle_bits,
+	.pin_interrupt_configure = gpio_intel_ehl_pin_interrupt_configure,
+};
+
+int gpio_intel_ehl_init(const struct device *dev)
+{
+	struct gpio_intel_ehl_data *data = dev->data;
+
+	DEVICE_MMIO_NAMED_MAP(dev, reg_base, K_MEM_CACHE_NONE);
+	data->pad_base = pad_base(dev);
+
+	__ASSERT(nr_isr_devs < GPIO_INTEL_EHL_NR_SUBDEVS, "too many subdevs");
+
+
+	if (nr_isr_devs == 0) {
+		/* Note that all controllers are using the same IRQ line.
+		 * So we can just use the values from the first instance.
+		 */
+		IRQ_CONNECT(DT_INST_IRQN(0),
+			    DT_INST_IRQ(0, priority),
+			    gpio_intel_ehl_isr, NULL,
+			    DT_INST_IRQ(0, sense));
+
+		irq_enable(DT_INST_IRQN(0));
+	}
+
+	isr_devs[nr_isr_devs++] = dev;
+
+	/* route to IRQ 14 */
+	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,
+			       MISCCFG_IRQ_ROUTE_POS);
+
+	return 0;
+}
+
+#define GPIO_INTEL_EHL_DEV_CFG_DATA(n)					       \
+	static const struct gpio_intel_ehl_config			       \
+		gpio_intel_ehl_cfg_##n = {				       \
+		.common = {						       \
+			.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(n),   \
+		},							       \
+		DEVICE_MMIO_NAMED_ROM_INIT(reg_base, DT_DRV_INST(n)),	       \
+									       \
+		.community_base = DT_INST_PROP(n, community_base),	       \
+		.group_index = DT_INST_PROP(n, group_index),		       \
+		.num_pins = DT_INST_PROP(n, ngpios),			       \
+		.pin_offset = DT_INST_PROP(n, pin_offset),		       \
+	};								       \
+									       \
+	static struct gpio_intel_ehl_data gpio_intel_ehl_data_##n;	       \
+									       \
+	DEVICE_DT_INST_DEFINE(n,					       \
+			      gpio_intel_ehl_init,			       \
+			      device_pm_control_nop,			       \
```

Comment:
Found it! This is causing the build error reported in CI. See related recent fix for other usage removed for same reason https://github.com/zephyrproject-rtos/zephyr/pull/34704

Reply:
Thanks, will update.